integration-tests/src/test/java/org/wso2/appserver/test/integration/TestSuiteListener.java

```diff
@@ -0,0 +1,181 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.appserver.test.integration;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.ISuite;
+import org.testng.ISuiteListener;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+
+public class TestSuiteListener implements ISuiteListener {
+
+    private static final Logger log = LoggerFactory.getLogger(TestSuiteListener.class);
+    private Process applicationServerProcess;
+    private int serverStartCheckTimeout;
+    private File appserverHome;
+    private int applicationServerPort;
+    private boolean isSuccessServerStartup = false;
+
+    @Override
+    public void onStart(ISuite iSuite) {
+
+        try {
+            log.info("Starting pre-integration test setup...");
+
+            appserverHome = new File(System.getProperty(TestConstants.APPSERVER_HOME));
+            log.info("Application server home : " + appserverHome.toString());
+
+            serverStartCheckTimeout = Integer.valueOf(System.getProperty(TestConstants.SERVER_TIMEOUT));
+
+
+            if (isPortAvailable(TestConstants.TOMCAT_DEFAULT_PORT)) {
+                log.info("Default port " + TestConstants.TOMCAT_DEFAULT_PORT + " is available.");
+                System.setProperty(TestConstants.APPSERVER_PORT, String.valueOf(TestConstants.TOMCAT_DEFAULT_PORT));
+                applicationServerPort = Integer.valueOf(System.getProperty(TestConstants.APPSERVER_PORT));
+            }
+
+
+            log.info("Starting the server...");
+            applicationServerProcess = startPlatformDependApplicationServer();
+
+            waitForServerStartup();
+
+            if (isSuccessServerStartup) {
```

Comment:
If server startup is not successful, we need to stop running the tests. Please test failure cases as well.